nlp-plugins/src/main/java/io/cdap/google/plugins/NLPConfig.java

```diff
@@ -0,0 +1,161 @@
+
+
+  @Name(PROPERTY_LANGUAGE_CODE)
+  @Description("Code of the language of the text data. E.g. en, jp, etc. If not provided" +
+    "Google Natural Language API will autodetect the language.")
+  @Macro
+  @Nullable
+  private String languageCode;
+
+  @Name(PROPERTY_ERROR_HANDLING)
+  @Description("Error handling strategy to use when there is an during NLP API call.")
+  private String errorHandling;
+
+  @Name(PROPERTY_SERVICE_ACCOUNT_FILE_PATH)
+  @Description("Path on the local file system of the service account key used "
+    + "for authorization. Can be set to 'auto-detect' when running on a Dataproc cluster. "
+    + "When running on other clusters, the file must be present on every node in the cluster.")
+  @Macro
+  @Nullable
+  protected String serviceFilePath;
+
+  public String getSourceField() {
+    return sourceField;
+  }
+
+  public NLPMethod getMethod() {
+    return Stream.of(NLPMethod.class.getEnumConstants())
+      .filter(keyType -> keyType.getValue().equalsIgnoreCase(methodName))
+      .findAny()
+      .orElseThrow(() -> new IllegalStateException(
+        String.format("Unsupported value for '%s': '%s'", PROPERTY_METHOD_NAME, methodName)));
+  }
+
+  public EncodingType getEncodingType() {
+    if (encoding == null) {
+      return EncodingType.NONE;
+    }
+
+    return Stream.of(EncodingType.class.getEnumConstants())
+      .filter(keyType -> keyType.toString().equals(encoding))
+      .findAny()
+      .orElseThrow(() -> new IllegalStateException(
+        String.format(
+          "Type of encoding specified '%s' is not supported. " +
+            "Supported values are NONE, UTF8, UTF16, UTF32.", encoding)));
+  }
+
+  @Nullable
+  public String getLanguageCode() {
+    return languageCode;
+  }
+
+  public ErrorHandling getErrorHandling() {
+    return Stream.of(ErrorHandling.class.getEnumConstants())
+      .filter(keyType -> keyType.getValue().equalsIgnoreCase(errorHandling))
+      .findAny()
+      .orElseThrow(() -> new IllegalStateException(
+        String.format("Unsupported value for '%s': '%s'", PROPERTY_ERROR_HANDLING, errorHandling)));
+  }
+
+  @Nullable
+  public String getServiceAccountFilePath() {
+    if (containsMacro(PROPERTY_SERVICE_ACCOUNT_FILE_PATH) || serviceFilePath == null ||
+      serviceFilePath.isEmpty() || serviceFilePath.equals(AUTO_DETECT)) {
+      return null;
+    }
+    return serviceFilePath;
+  }
+
+  public void validate(FailureCollector failureCollector, Schema inputSchema) {
+    if (inputSchema.getField(sourceField) == null) {
+      failureCollector.addFailure(String.format("Field '%s' does not exist in input schema", sourceField), null)
+        .withConfigProperty(PROPERTY_SOURCE_FIELD);
+    }
+
+    // trigger getters, so that they fail if value cannot be converted to enum.
+    try {
+      getErrorHandling();
+    } catch (IllegalStateException ex) {
+      failureCollector.addFailure(ex.getMessage(), null)
+        .withStacktrace(ex.getStackTrace())
```

Comment:
don't need the stacktrace here or for the method and encoding type because we know exactly what went wrong. The stack trace is more for handling unknown exceptions where we don't know exactly what went wrong, usually because we're calling some library.